add-on/src/lib/ipfs-request.js

```diff
@@ -527,8 +536,8 @@ function isRecoverable (request, state, ipfsPathValidator) {
   return state.recoverFailedHttpRequests &&
     request.type === 'main_frame' &&
     (recoverableNetworkErrors.has(request.error) || recoverableHttpError(request.statusCode)) &&
-    ipfsPathValidator.publicIpfsOrIpnsResource(request.url) &&
-    !request.url.startsWith(state.pubGwURLString)
+    (ipfsPathValidator.publicIpfsOrIpnsResource(request.url) || ipfsPathValidator.ipfsOrIpnsSubdomain(request.url)) &&
+    !request.url.startsWith(state.pubGwURLString) && !request.url.includes(state.subdomainGwURL.hostname)
```

Comment:
We already have `ipfsPathValidator.publicIpfsOrIpnsResource`, let's move those custom subdomain checks to similar `publicIpfsOrIpnsSubdomainResource`. Will be easier to maintain and reuse.

Reply:
I'd love to, and left a todo above the `ipfsOrIpnsSubdomainResource` in the original PR. Right now we don't have a `subdomainGwUrl` state variable that allows for a test similar to the one we have in `publicIpfsOrIpnsSubdomainResource`, short of hard-coding one. Glad to hear if there's another way.

Reply:
Ah yes, we don't have localhost subdomains yet (https://github.com/ipfs/in-web-browsers/issues/109).
You are right, better to add those methods when that happens.